src/main/kotlin/net/thesilkminer/mc/ematter/common/feature/sbg/SbgTileEntity.kt

```diff
@@ -0,0 +1,200 @@
+package net.thesilkminer.mc.ematter.common.feature.sbg
+
+import net.minecraft.block.Block
+import net.minecraft.nbt.NBTTagCompound
+import net.minecraft.tileentity.TileEntity
+import net.minecraft.util.EnumFacing
+import net.minecraft.util.ITickable
+import net.minecraftforge.common.capabilities.Capability
+import net.thesilkminer.kotlin.commons.lang.uncheckedCast
+import net.thesilkminer.mc.boson.api.direction.Direction
+import net.thesilkminer.mc.boson.api.energy.Holder
+import net.thesilkminer.mc.boson.api.energy.Producer
+import net.thesilkminer.mc.boson.prefab.energy.holderCapability
+import net.thesilkminer.mc.boson.prefab.energy.producerCapability
+import net.thesilkminer.mc.ematter.common.shared.TemperatureSources
+import kotlin.math.roundToInt
+
+@Suppress( "EXPERIMENTAL_API_USAGE" , "EXPERIMENTAL_OVERRIDE" , "EXPERIMENTAL_UNSIGNED_LITERALS" )
+class SbgTileEntity : TileEntity(), Producer, Holder, ITickable {
+
+    internal companion object {
+        private const val ROOM_TEMPERATURE: Int = 295
+        //that's the storage of the basic mad
+        private const val MAX_CAPACITY: ULong = 3_141uL
+
+        //next one is here in case you want to make this changeable through a config file in the future
+        private const val POWER_MULTIPLIER: Double = 1.0
+
+        private const val THRESHOLD_TIME: UInt = 60u
+        private const val BURST_TIME: UInt = 40u
+
+        private const val POWER_KEY = "power"
+        private const val THRESHOLD_KEY = "threshold_time"
+        private const val BURST_KEY = "burst_time"
+    }
+
+    //temperature fields
+    private var tempDifference: UInt = 0u
+    private var effectiveness: Double = 1.0
+
+    //timer fields
+    private var threshold: UInt = THRESHOLD_TIME
+    private var burstTimer: UInt = BURST_TIME
+
+    //power fields
+    private var powerProduced: ULong = 0uL
+    private var powerStored: ULong = 0uL
+
+    override val producedPower: ULong get() = this.powerProduced
+    override val storedPower: ULong get() = this.powerStored
+    override val maximumCapacity: ULong get() = MAX_CAPACITY
+
+    /**
+     * Tries every tick to generate power. Only does so if this is a server instance, the [tempDifference] is not 0
+     * and the [threshold] is at 0.
+     */
+    override fun update() {
+        if( !super.world.isRemote ) {
+            if( this.tempDifference != 0u ) {
+                if( this.threshold > 0u ) {
+                    this.threshold--
+                }
+                else {
+                    if( this.storedPower < MAX_CAPACITY ) {
+                        this.generatePower()
+                    }
+                }
+            }
```

Comment:
You aren't pushing the produced energy anywhere. Is this intended for a later addition when cables are actually a thing or was it an oversight?

Reply:
Actually I haven't thought about that at all. But I guess this can be added later on when cables or such are a thing?
I'll look into that and try to figure out how to push it somewhere and if I can't I'll ask you again later.